spec/docs/logicalSource.md

```diff
@@ -0,0 +1,11 @@
+# Defining Logical Sources
+
+A <dfn>logical source</dfn> is a function on a [=data source=] that provides data to be mapped to [=RDF triples=].
+
+A [=logical source=] (`rml:LogicalSource`) MUST have:
+* exactly one `rml:referenceFormulation` property, whose value is a <dfn>reference formulation</dfn> which defines how the underlying [=data source=] is to be accessed, and which [=expressions=] can be evaluated on [=logical iterations=],
+* zero or one `rml:iterator` property, whose value is a <dfn data-lt="iterator">logical iterator</dfn> that defines a sequence of [=logical iterations=] on the [=data source=]. If no [=iterator=] is provided, a <dfn class="lint-ignore">default iterator</dfn> MUST be associated with the [=reference formulation=].
```

Comment:
We need to link to rml-io-registry here for the default iterator and how expressions are formed per reference formulation.

Reply:
I think it should be the other way around. rml-io-registry should link to core and use the definitions from core (and io). However, it would be good to have some informal links from core to relevant documents like rml-io-registry.

Reply:
Sure, but we need to link in some way otherwise non-experts will be confused.
I would say in the spec that the details of how a reference formulation is handled and implemented for each data format is specified in rml-io-registry.